homeassistant/components/sensorpush_cloud/sensor.py

```diff
@@ -0,0 +1,180 @@
+        translation_key="atmospheric_pressure",
+        native_unit_of_measurement=UnitOfPressure.INHG,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_BATTERY_VOLTAGE,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_registry_enabled_default=False,
+        translation_key="battery_voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_DEWPOINT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_registry_enabled_default=False,
+        translation_key="dewpoint",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_SIGNAL_STRENGTH,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_registry_enabled_default=False,
+        translation_key="signal_strength",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_VAPOR_PRESSURE,
+        device_class=SensorDeviceClass.PRESSURE,
+        entity_registry_enabled_default=False,
+        translation_key="vapor_pressure",
+        native_unit_of_measurement=UnitOfPressure.KPA,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SensorPushCloudConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SensorPush Cloud sensors."""
+    coordinator: SensorPushCloudCoordinator = entry.runtime_data
+    async_add_entities(
+        [
+            SensorPushCloudSensor(coordinator, entity_description, device_id)
+            for entity_description in itertools.chain(SENSORS, DISABLED_SENSORS)
+            for device_id in coordinator.device_ids
+        ]
+    )
+
+
+class SensorPushCloudSensor(
+    CoordinatorEntity[SensorPushCloudCoordinator], SensorEntity
+):
+    """SensorPush Cloud sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SensorPushCloudCoordinator,
+        entity_description: SensorEntityDescription,
+        device_id: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self.device_id = device_id
+
+        if device_id not in self.coordinator.devices:
+            LOGGER.warning(f"Ignoring inactive device: {device_id}")
+            self._attr_available = False
+            return
+
+        device = self.coordinator.devices[device_id]
+        self._attr_unique_id = f"{device[ATTR_DEVICE_ID]}_{entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_BLUETOOTH, device[ATTR_BLUETOOTH])},
```

Comment:
The API is via cloud, not bluetooth. I am not sure if you can use the bluetooth connection for that if the link is not native bluetooth.

Reply:
I went back and forth on this. It looks like this field is only used to establish a unique identity for a given device. The cloud API provides the MAC address for connected Bluetooth sensors, which is the only external attribute a user could use to disambiguate sensors. That said, I agree it could be surprising if this attribute were used in a different way down the road - I'll remove this.